File: frontend/src/utils/identity.ts
/**
 * MockInterview.ai — Identity Utility
 * Handles persistent JudgeID generation and retrieval using localStorage.
 */

const SESSION_ID_KEY = 'mockinterview-session-id';

/**
 * Gets the persistent JudgeID for this browser.
 * Generates a new one if it doesn't exist.
 */
export function getSessionId(): string {
  let id = localStorage.getItem(SESSION_ID_KEY);
  
  if (!id) {
    id = crypto.randomUUID();
    localStorage.setItem(SESSION_ID_KEY, id);
  }
  
  return id;
}

/**
 * Formats a name to be used for Google Cloud / Gemini resources.
 * Ensures it follows naming conventions (lowercase, alphanumeric, etc).
 */
export function getResourcePrefix(): string {
  const id = getSessionId();
  // We use the first 8 chars + some entropy or just the clean ID
  // Gemini store names must start with a letter and contain only letters, numbers, hyphens
  const cleanId = id.replace(/[^a-zA-Z0-9]/g, '').toLowerCase();
  return `mock-interview-store-${cleanId.slice(0, 12)}`;
}

File: app/app_utils/identity.py
"""
MockInterview.ai — Identity & Naming Utility (Backend)
Ensures consistent naming for Gemini FileSearchStores based on User/Judge IDs.
"""

import re

def get_store_name_for_user(user_id: str) -> str:
    """
    Formats a Gemini FileSearchStore name based on the user's UUID.
    Naming rules for Gemini stores:
    - Must start with a letter.
    - Can only contain lowercase letters, numbers, and hyphens.
    - Max length usually 64 chars.
    """
    # Clean the UUID: remove non-alphanumeric, lowercase
    clean_id = re.sub(r'[^a-zA-Z0-9]', '', user_id).lower()
    
    # We prefix to identify our app's resources
    return f"mock-interview-store-{clean_id[:12]}"

def get_display_name_for_user(user_id: str) -> str:
    """Friendly display name for the store in the Google AI Studio / GCP console."""
    return f"MockInterview AI Store ({user_id[:8]})"
